### PR TITLE
Task 1

### PR DESCRIPTION
**Pull request by AlenaKharytonchyk**
Task 1
Create package.json file and install React, Redux, React-Redux, React-Router,
Jest. Install and configure webpack & babel to get build artifact by running npm command.
Write components implementing HTML markup for required design (see images at the
beginning of the document). For this part, no need to implement API calls and routing, the
task can be done with mocked data.
Use <ErrorBoundary> component for catching and displaying errors
https://reactjs.org/docs/error-boundaries.html. You could create one component and wrap
all your application, or use several components.

Evaluation criteria*
2 Installed Webpack, Markup is done
with React Components
3
Use <ErrorBoundary>
component for catching and
displaying errors
4 Use smart/dumb components
approach
5 100% decomposition
(evaluated by mentor)

Everything is done according to the evaluation criteries

### CODE REVIEW

File: my-app/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "react": "^16.12.0",
+    "react-dom": "^16.12.0",
+    "react-redux": "^7.1.3",
+    "react-router-dom": "^5.1.2",
+    "react-scripts": "3.3.0",
+    "redux": "^4.0.4",
+    "redux-thunk": "^2.3.0"
+  },
+  "scripts": {
+    "start": "react-scripts start",
```
Comment (NikitaHapeyeu) [inline my-app/package.json]: we have to use our npm command, not use create-react-app
Please see the requirements

File: my-app/src/Components/header/header.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import './header.scss';
+
+const Header = () => {
+  return (
+    <header>
+
```
Comment (NikitaHapeyeu) [inline my-app/src/Components/header/header.jsx]: is a need here free space?

File: my-app/src/Components/notFound/notFound.jsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import './notFound.scss';
+
+const NoResults = (props) => {
+  const {result_amount} = props;
+  return(
+    <span className='not-found'>No films found</span>
```
Comment (NikitaHapeyeu) [inline my-app/src/Components/notFound/notFound.jsx]: please use " instead ' in className

File: my-app/src/Components/panicBtn/panicButton.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isClicked: false
+        };
+        this.onButtonClick = this.onButtonClick.bind(this);
+    }
+
+    onButtonClick() {
```
Comment (NikitaHapeyeu) [inline my-app/src/Components/panicBtn/panicButton.jsx]: read more about method syntax like a `onButtonClick = () => {....`
it is not required binding on `this`

File: my-app/src/pages/Movie_info.jsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import CardContainer from "../Components/cardContainer/cardContainer";
+import MovieCard from "../Components/movieCard/movieCard";
+import SearchResultBy from "../Components/searchResult_forMovie_page/searchResultBy";
+import PageContainer from "../Components/headerFooter/headerFooter";
+
+function MovieInfo({movies}) {
+  const [movie] = movies;
```
Comment (NikitaHapeyeu) [inline my-app/src/pages/Movie_info.jsx]: Please apply arrow function

File: my-app/src/pages/Search.jsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import SearchForm from "../Components/searchForm/searchForm";
+import TitleText from "../Components/titleText/title";
+import SearchBy from "../Components/searchBy/search";
+import CardContainer from "../Components/cardContainer/cardContainer";
+import SearchResultString from "../Components/searchResultString/searchResult";
+import PageContainer from "../Components/headerFooter/headerFooter";
+
+function SearchPage({movies}) {
+
```
Comment (NikitaHapeyeu) [inline my-app/src/pages/Search.jsx]: I suggest to use arrow function

File: my-app/package.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    "build": "react-scripts build",
```
Comment (NikitaHapeyeu) [inline my-app/package.json]: also we have to create webpack config
image

**Comment by NikitaHapeyeu** [comment]
I guess that needs to think about file structure - separate components on containers(smart), shared(reuseable) and presentational(dumb).

File: my-app/src/Components/header/header.jsx (hunk 2 above)
Comment (AlenaKharytonchyk) [reply my-app/src/Components/header/header.jsx]: @NikitaHapeyeu This component is redundant. Deleted

File: my-app/src/Components/notFound/notFound.jsx (hunk 3 above)
Comment (AlenaKharytonchyk) [reply my-app/src/Components/notFound/notFound.jsx]: @NikitaHapeyeu Done

File: my-app/src/pages/Movie_info.jsx (hunk 5 above)
Comment (AlenaKharytonchyk) [reply my-app/src/pages/Movie_info.jsx]: @NikitaHapeyeu Done

File: my-app/src/pages/Search.jsx (hunk 6 above)
Comment (AlenaKharytonchyk) [reply my-app/src/pages/Search.jsx]: @NikitaHapeyeu Done

File: my-app/package.json (hunk 7 above)
Comment (AlenaKharytonchyk) [reply my-app/package.json]: @NikitaHapeyeu Done

File: my-app/package.json (hunk 1 above)
Comment (AlenaKharytonchyk) [reply my-app/package.json]: @NikitaHapeyeu Done

**Pull request merged**